### PR TITLE
 Support EC Key-Imports for the JavaKeystoreKeyProvider #26936 

### LINKED ISSUE #26936: Support EC Key-Imports for the JavaKeystoreKeyProvider
### Description

Currently, its only possible to import keys of type RSA into Keycloak, using the  [JavaKeystoreKeyProvider](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java). In order to support the various signature methods for [OID4VC](https://github.com/keycloak/keycloak/issues/25936), it would be good to also support the import of EC keys.  

### Discussion

_No response_

### Motivation

To properly support Verifiable Credentials and the usage of stable issuer did's, a method to import EC keys into Keycloak is required.

### Details

_No response_

### PR DESCRIPTION
**Pull request by wistefan**
Adds support for the import of EC Keys with the JavaKeystoreKeyProvider.
closes https://github.com/keycloak/keycloak/issues/26936

### CODE REVIEW

File: services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java [hunk 1]
```diff
@@ -43,46 +48,57 @@
...
-            X509Certificate certificate = (X509Certificate) keyStore.getCertificate(keyAlias);
-            if (certificate == null) {
-                certificate = CertificateUtils.generateV1SelfSignedCertificate(keyPair, realm.getName());
-            }
-
-            KeyUse keyUse = KeyUse.valueOf(model.get(Attributes.KEY_USE, KeyUse.SIG.getSpecName()).toUpperCase());
-
-            return createKeyWrapper(keyPair, certificate, loadCertificateChain(keyStore, keyAlias), keyUse);
+            return switch (algorithm) {
+                case Algorithm.HS256, Algorithm.HS384, Algorithm.HS512, Algorithm.RS256, Algorithm.RS384, Algorithm.RS512 ->
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java:96]: I suppose the `HS*` algorithms here should be replaced by `PS*` algorithms on this line? As I see that `Attributes.RS_ALGORITHM_PROPERTY` supports keys of type `Algorithm.RS256, Algorithm.RS384, Algorithm.RS512, Algorithm.PS256, Algorithm.PS384, Algorithm.PS512` but it does not support `Algorithm.HS256, Algorithm.HS384, Algorithm.HS51` .

**Review by mposolda — COMMENTED** [review]
@wistefan Thanks!

I've added one comment inline, could you please check it?

File: services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java (hunk 1 above)
Comment (wistefan) [reply services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProvider.java]: You're right, I fixes it. see https://github.com/keycloak/keycloak/pull/27030/commits/19e7795fc24caec6780afc0d5beaed8c0257a3d4

**Review by mposolda — APPROVED** [review]
@wistefan Thanks!

**Pull request merged**